db/utils.go

```diff
@@ -0,0 +1,30 @@
+package db
+
+import (
+	"time"
+
+	"github.com/couchbase/sync_gateway/base"
+	"golang.org/x/net/context"
+)
+
+type BackgroundTaskFunc func(ctx context.Context) error
+
+// backgroundTask runs task at the specified time interval in its own goroutine until stopped
+func NewBackgroundTask(name string, contextID string, task BackgroundTaskFunc, interval time.Duration, c chan bool) {
+	ctx := context.WithValue(context.Background(), base.LogContextKey{}, base.LogContext{CorrelationID: base.NewChangeCacheContextID(contextID)})
+	base.InfofCtx(ctx, base.KeyAll, "Created background task: %q with interval %v", name, interval)
+	go func() {
+		for {
+			select {
+			case <-time.After(interval):
+				base.DebugfCtx(ctx, base.KeyAll, "Running background task: %q", name)
+				if err := task(ctx); err != nil {
```

Comment:
As discussed, let's change error handling such that if a background task returns error, it indicates that we want to terminate the scheduled task.  We don't have a use case like that today, so we should change the compact background task to just warn on error and return nil.